### ISSUE TITLE
`Combined Scheme` should be highlighted

### ISSUE THREAD

**Issue by JasonGross** [ProofGeneral/PG#73]
![image](https://cloud.githubusercontent.com/assets/396076/15098602/c91fd2de-150e-11e6-9972-592bd08d7913.png)

**Comment by cpitclaudel** [comment]
Yeah, see my recent email to Coqdev about this. I think we're fighting a loosing battle here. Coq has a list of keywords; we just need a way to export it and load it when Coq starts.

**Comment by psteckler** [comment]
There's actually a mechanism in PG for a prover to configure an Elisp
variable: proof-shell-repl-set-elisp-variable-regexp. You could use that to
set a list of highlight-able syntax items inside PG.

Yes, the mechanism is a backdoor. I didn't tell you about this mechanism.

-- Paul

**Comment by cpitclaudel** [comment]
Do you mean `proof-shell-set-elisp-variable-regexp`? Sure, but we still need Coq to issue the configuration calls, right?

**Comment by psteckler** [comment]
That's the right name (I had temporarily renamed the variable in my copy).

Coq would have to expose the names, preferably in response to a query,
rather than just offering them on startup.

-- Paul

**Comment by cpitclaudel** [comment]
It doesn't currently offer them at all, though, right?
It would be very nice to have a command that prints a summary of what the IDE needs to configure itself, indeed :)

**Comment by psteckler** [comment]
No, as far as I know.

-- Paul

**Comment by JasonGross** [comment]
Perhaps you are looking for `Print Grammar vernac`? ;)  (`pr_grammar` in `toplevel/metasyntax.ml` suggests that the known grammars are `constr` (equivalent to `operconstr` and `binder_constr`), `pattern`, `tactic`, and `vernac`.)

**Comment by cpitclaudel** [comment]
No, not quite; company-coq already parses the output of `Print Grammar tactic`, but it's a mess. In particular, the parser is much too complex (and the format much too brittle) to seriously rely on it.

**Comment by JasonGross** [comment]
Can't you approximate it by looking for a sequence of `IDENT`s and only highlighting the initial sequences of `IDENT`s?

**Comment by cpitclaudel** [comment]
Yup; that's pretty much what I call hacky :)
We could also just take the list that company-coq extracts from the manual: https://github.com/cpitclaudel/company-coq/blob/master/company-coq-abbrev.el#L526 (`Combined Scheme` was in there, for example)
